src/search/landmarks/landmark_heuristic.cc

```diff
@@ -70,6 +79,42 @@ void LandmarkHeuristic::initialize(const plugins::Options &opts) {
     }
 }
 
+bool LandmarkHeuristic::landmark_graph_has_cycle_of_natural_orderings() {
+    int num_landmarks = lm_graph->get_num_landmarks();
+    vector<bool> closed(num_landmarks, false);
+    vector<bool> visited(num_landmarks, false);
+    for (auto &node : lm_graph->get_nodes()) {
+        if (depth_first_search_for_cycle_of_natural_orderings(
+                *node, closed, visited)) {
+            return true;
+        }
+    }
+    return false;
```

Comment:
Is this equivalent to the old code? It's wasteful because it is O(N^2) even though the problem has simple O(N) algorithms.

If the old code did this too, we can leave it as is, but I would at least add a TODO and add this as an open item in our landmark metaissue.

Reply:
(...or perhaps it is already O(N) because of the way closed and visited are handled by the called function as two-way arguments, which I didn't check. That's probably the case. In that case the code is perhaps not as pretty as it could be, but OK performance-wise.)